core/templates/pages/contributor-dashboard-page/contributor-badges/contributor-badges.component.ts

```diff
@@ -103,11 +103,16 @@ export class ContributorBadgesComponent {
     if (userContributionRights === null) {
       throw new Error('Cannot fetch user contribution rights.');
     }
-    userContributionRights.can_review_translation_for_language_codes.map(
-      (languageCode) => {
-        const languageDescription = this.languageUtilService
-          .getAudioLanguageDescription(
-            languageCode);
+
+    const allContributionStats = await this
+      .contributionAndReviewStatsService.fetchAllStats(username);
+
+    if (allContributionStats.translation_review_stats.length > 0) {
```

Comment:
Why not do this in the `if allContributionStats.translation_review_stats.length > 0)` block below?

Reply:
Here we initialize `totalTranslationStats` for available languages. Later on we increment stats for each contribution.

Reply:
Yeah, but you can combine the two blocks. Initialize `totalTranslationStats` if needed, otherwise increment the counts.